python/lsst/obs/base/butler_tests.py

```diff
@@ -130,7 +133,17 @@ def setUp_butler_get(
             raw_header_wcs=raw_header_wcs,
         )
 
+    def _require_gen2(self):
+        if isinstance(self.butler, Butler):
+            self.skipTest("This test requires daf_persistence Butler, not daf_butler Butler.")
+
+    def _is_gen3(self):
+        if isinstance(self.butler, Butler):
```

Comment:
Is it worth flipping this around to be `is_gen2` and call `isinstance(self.butler, lsst.daf.persistence.Butler)` instead, so that we have a very clear thing to remove when we finally remove daf_persistence?